pyemma/msm/estimators/maximum_likelihood_msm.py

```diff
@@ -1361,3 +1361,498 @@ def OOM_sigma(self):
+        # Convergence flag for hatpi
+        self._estimated = False
+        self._max_iter = maxiter
+        self.debug = debug
+        self._max_cache = max_cache
+        self._is_estimated = False
+
+    def _log_likelihood_biased(self, C, T, E, mhat, ws):
+        """
+          Evaluate AMM likelihood.
+        """
+        ll_unbiased = msmest.log_likelihood(C, T)
+        ll_bias = -_np.sum(ws*(mhat-E)**2.)
+        return ll_unbiased + ll_bias
+
+    def _update_G(self):
+        """
+          Update G, observable covariance.
+          See SI of [1].
+        """
+        _tmp = self.E_active*self.pihat[:, None]
+        self._G = _np.dot(self.E_active.T, self.E_active*self.pihat[:, None])-self.mhat[:, None]*self.mhat[None, :]
+
+    def _update_Q(self):
+        """
+          Compute Q, a weighted sum of the R-tensor.
+          See SI of [1].
+        """
+        self.Q = _np.zeros((self.n_mstates_active, self.n_mstates_active))
+        for k in range(self.n_exp_active):
+          self.Q = self.Q + self.w[k]*self._S[k]*self._get_Rk(k)
+        self.Q = -2.*self.Q
+
+    def _update_Rslices(self, i):
+        """
+          Computation of multiple slices of R tensor.
+          When _estimate(.) is called the R-tensor is split into segments whose maximum size is
+          specified by max_cache argument (see constructor).  
+          _Rsi specifies which of the segments are currently in cache.
+          For equations check SI of [1].
+        """
+        pek = self.pihat[:, None]*self.E_active[:,i*self._slicesz:(i+1)*self._slicesz]
+        pp = (self.pihat[:, None] + self.pihat[None, :])
+        ppmhat = pp*self.mhat[i*self._slicesz:(i+1)*self._slicesz, None, None]
+        self._Rs=(pek[:, None,:]+pek[None, :, :]).T - ppmhat
+        self._Rsi = i
+
+    def _get_Rk(self, k):
+        """
+          Convienence function to get cached value of an Rk slice of the R tensor.
+          If we are outside cache, update the cache and return appropriate slice.
+        """
+        if k>(self._Rsi+1)*self._slicesz or k<(self._Rsi)*self._slicesz:
+          self._update_Rslices(_np.floor(k/self._slicesz).astype(int))
+          return self._Rs[k%self._slicesz]
+        else:
+          return self._Rs[k%self._slicesz]
+
+    def _update_pihat(self):
+        """
+          Update stationary distribution estimate of Augmented Markov model (\hat pi) 
+        """ 
+        expons = (self.lagrange[:, None]*self.E_active.T).sum(axis=0)
+        expons = expons - expons.max()
+
+        _ph_unnom = self.pi*_np.exp(expons)
+        self.pihat = (_ph_unnom/_ph_unnom.sum()).reshape(-1,)
+
+    def _update_mhat(self): 
+        """
+          Updates mhat (expectation of observable of the Augmented Markov model) 
+        """ 
+        self.mhat = _np.dot(self.pihat.reshape((self.n_mstates_active,)), self.E_active[:]) 
+        self._update_S() 
+
+    def _update_S(self):
+        """
+          Computes slope in observable space.
```

Comment:
In this case you can write 

""" Computes slope in observable space. """

If you have text, then use

""" Title

blabla
(empty line here)
"""